src/Logging.chpl

```diff
@@ -0,0 +1,88 @@
+module Logging {
+    use Set;
+    use IO;
+    use DateTime;
+
+    /*
+     * The LogLevel enum is used to provide a strongly-typed means of
+     * configuring the logging level of a Logger object
+     */
+    enum LogLevel {DEBUG,INFO,WARN,ERROR,CRITICAL};
+    
+    /*
+     * The Logger class provides structured log messages at various levels
+     * of logging sensitivity analogous to other languages such as Python.
+     */
+    class Logger {
+        var level = LogLevel.WARN;
+        
+        var warnLevels = new set(LogLevel,[LogLevel.WARN, LogLevel.INFO,
+                           LogLevel.DEBUG]);
+        
+        var criticalLevels =  new set(LogLevel, [LogLevel.CRITICAL, LogLevel.WARN,
+                                         LogLevel.INFO,LogLevel.DEBUG]);
+                                         
+        var errorLevels =  new set(LogLevel,[LogLevel.ERROR, LogLevel.CRITICAL,
+                           LogLevel.WARN, LogLevel.INFO, LogLevel.DEBUG]);
+                             
+        var infoLevels = new set(LogLevel,[LogLevel.INFO,LogLevel.DEBUG]);  
+        
+        var printDate: bool = true;
+        
+        proc debug(moduleName, routineName, lineNumber, msg: string) throws {
+            if level == LogLevel.DEBUG  {
+                writeln(generateLogMessage(moduleName, routineName, lineNumber, 
+                                            msg, "DEBUG"));
+                stdout.flush();
+            }
+        }
+        
+        proc info(moduleName, routineName, lineNumber, msg: string) throws {
+            if infoLevels.contains(level) {
```

Comment:
Another micro-optimization, but these `contains` calls aren't free, and it should be faster to just check against all the enums instead of putting them in a set. Or if the enums are ordered I think you can just compare them. Something like:

```chpl
if level >= LogLevel.info {
  ...
}
```